### PR TITLE
permissions-center: implement permissions sync job search by repo name.

### PR DESCRIPTION
**Pull request by sashaostrikov**
Test plan:
GraphQL tests added.
Database layer unit tests added.

First part of https://github.com/sourcegraph/sourcegraph/issues/47303

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (+0.13 kb) | 0.00% (+0.13 kb) | 0.00% (0.00 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/ao/pc/search-by-repo-name/compare-report.html;tab=live_object) for a full comparison between the commits 2a4ca15 and 15ef2f5 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: internal/database/permission_sync_jobs.go [hunk 1]
```diff
@@ -486,12 +497,20 @@ func (opts ListPermissionSyncJobOpts) sqlConds() []*sqlf.Query {
 	if opts.NotCanceled {
 		conds = append(conds, sqlf.Sprintf("cancel = false"))
 	}
+
+	if opts.Query != "" {
+		if opts.SearchType == PermissionsSyncSearchTypeRepo {
```
Comment (thenamankumar) [inline internal/database/permission_sync_jobs.go:502]: It should rather be:

```golang
if opts.SearchType == PermissionsSyncSearchTypeRepo {
  conds = append(conds, sqlf.Sprintf("permission_sync_jobs.repository_id IS NOT NULL"))

  if opts.Query != "" {
    conds = append(conds, sqlf.Sprintf("repo.name ILIKE %s", "%"+opts.Query+"%"))
  }
}
```

Same should be followed when you add for users. 

Selecting Repo in the filter itself should only display repo based sync jobs, even when the query is empty. 

Query should additionally filter those repo sync jobs by repo name.

File: internal/database/permission_sync_jobs_test.go [hunk 2]
```diff
@@ -184,6 +184,21 @@ func TestPermissionSyncJobs_CreateAndList(t *testing.T) {
...
+			opts:     ListPermissionSyncJobOpts{Query: "where's the search type, Lebowski?"},
+			wantJobs: jobs,
+		},
+		{
+			name:     "Repo name search, case-insensitivity",
+			opts:     ListPermissionSyncJobOpts{Query: "TeST", SearchType: PermissionsSyncSearchTypeRepo},
+			wantJobs: []*PermissionSyncJob{jobs[0], jobs[3]},
+		},
+		{
+			name:     "Repo name search",
```
Comment (thenamankumar) [inline internal/database/permission_sync_jobs_test.go:198]: Add tests to cover case when Query is empty but SearchType is `repo`

Comment (thenamankumar) [reply internal/database/permission_sync_jobs_test.go:198]: ^ it should return only repo based sync jobs.

File: internal/database/permission_sync_jobs.go (hunk 1 above)
Comment (sashaostrikov) [reply internal/database/permission_sync_jobs.go:502]: aha, I thought the other way around, will fix it now, then

**Review by thenamankumar — APPROVED** [review]

**Pull request merged**